pkg/git/git.go

```diff
@@ -269,6 +269,16 @@ func updateRepo(repoPath string) (*Repo, error) {
 	}
 
 	// Update the repo
+	if err := command.NewWithWorkDir(
+		repoPath, gitExecutable, "stash", "--include-untracked",
+	).RunSilentSuccess(); err != nil {
+		return nil, errors.Wrap(err, "unable to stash changes")
+	}
+	if err := command.NewWithWorkDir(
```

Comment:
The state of the currently checked out branch might be important for the user. So I’m not in favor of checking out master generally on update. I suggest in moving that part to the release notes implementation.

Reply:
> The state of the currently checked out branch might be important for the user.

It's a fair point. In a [comment below](https://github.com/kubernetes/release/pull/1126#discussion_r382913566) I thought about how we might do this differently to preserve local repo state after we're done.

> I suggest in moving that part to the release notes implementation.

We might have to refactor this code to do that. The release notes code calls `CloneOrOpenGitHubRepo`, which calls `CloneOrOpenRepo`, which upon discovering the repo already exists will call `updateRepo`.

If we want the update repo behaviour to be different for `krel release-notes` (i.e, don't just `git pull` on whatever branch is there) we're going to either have to call the clone directly from the release notes code and handle the update ourselves in there, or we're going to need a flag or something passed into the `CloneOrOpenRepo`.

Reply:
Alright I see, let’s merge that discussion with https://github.com/kubernetes/release/pull/1126#discussion_r382915852